FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : ScalarValue<string>
+    {
+        /// <summary>
+        /// Create a Ref from a string, such as <c>new Ref("databases/prydain")</c>.
+        /// </summary>
+        /// <param name="value">Value.</param>
+        public RefV(string value) : base(value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer) =>
+            writer.WriteObject("@ref", Value);
+    }
+
+    public abstract class ObjectRef : RefV
```

Comment:
So, I imagine that in C#, a call to the ref function to create structured refs would look something like:

```
Ref(Class("my-class"), "123")
```

If you need to manually create a structured ref value, I guess the most natural translation would be something like:

```
new RefV(new ClassV("my-class"), "123")
```

If that is the most natural path, I guess I would still ague about the class names as of now. `ObjectInstanceRef`, `CustomClassRef`, they sound a lot more scary than `new RefV(id: '123", class: new ClassV("some-class"))`.